### PR TITLE
[JENKINS-72854] UX improvement: make pod templates readonly from UI for NonConfigurableKubernetesCloud

### PR DESCRIPTION
**Pull request by Pldi23**
https://issues.jenkins.io/browse/JENKINS-72854

This issue is just a UX issue, backend do not save changes in pod templates for NonConfigurableKubernetesCloud (covered by NonConfigurableKubernetesCloudTest#configRoundTrip). But users are very confused as they may assume that changes should be saved.

### Testing done

I tested manually by running NonConfigurableKubernetesCloudTest#configRoundTrip, I stopped the test in debug mode and verified that for NonConfigurableKubernetesCloud Pod templates can not be edited anymore from UI, user see the message `This Pod template is associated with a non-configurable cloud setup.` , "Save" button is not visible and "Delete Pod Template" on side panel is not visible. 
For KubernetesCloud no changes on UX (it is still configrable).

Screenshot 2024-06-03 at 12 34 38

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by Vlatombe** [comment]
There is a built-in mechanism in core to switch all controls to readonly
```
<j:set var="readOnlyMode" value="${instance.readonlyFromUi}"/>
```

**Comment by Pldi23** [comment]
> There is a built-in mechanism in core to switch all controls to readonly
> 
> ```
> <j:set var="readOnlyMode" value="${instance.readonlyFromUi}"/>
> ```

Yeah, I tried to use it but [enum dropdown](https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/config.jelly#L40-L42) is still active with it.
Screenshot 2024-05-30 at 10 17 50

**Comment by jglick** [comment]
So fix `enum.jelly` in core?

**Comment by Pldi23** [comment]
> So fix `enum.jelly` in core?

Ok, I agree it's more rational. Let's fix this in core.

**Review by aneveux — APPROVED** [review]
LGTM

**Comment by jglick** [comment]
Should this be closed in favor of https://github.com/jenkinsci/jenkins/pull/9336?

**Comment by Pldi23** [comment]
> Should this be closed in favor of [jenkinsci/jenkins#9336](https://github.com/jenkinsci/jenkins/pull/9336)?

No, we need both: current one and https://github.com/jenkinsci/jenkins/pull/9336

**Comment by Pldi23** [comment]
@Vlatombe , @jglick If you are ok with this change could you please merge the PR (I don't have a required permissions).

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/index.jelly [hunk 1]
```diff
@@ -29,14 +29,20 @@ THE SOFTWARE.
       <f:form method="post" action="configSubmit" name="config" class="jenkins-form">
         <j:set var="instance" value="${it}"/>
         <j:set var="descriptor" value="${instance.descriptor}"/>
+        <j:set var="readOnlyMode" value="${instance.readonlyFromUi == true}"/>
+        <j:if test="${instance.readonlyFromUi == true}">
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/index.jelly:33]:
```suggestion
        <j:set var="readOnlyMode" value="${instance.readonlyFromUi}"/>
        <j:if test="${instance.readonlyFromUi}">
```

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/index.jelly [hunk 2]
```diff
@@ -29,14 +29,20 @@ THE SOFTWARE.
...
+        </j:if>
 
         <!-- main body of the configuration -->
         <st:include it="${instance}" page="config.jelly"/>
 
         <l:hasAdministerOrManage>
-          <f:bottomButtonBar>
-            <f:submit value="${%Save}"/>
-          </f:bottomButtonBar>
+          <j:if test="${instance.readonlyFromUi != true}">
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/index.jelly:41]:
```suggestion
          <j:if test="${!instance.readonlyFromUi}">
```

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/sidepanel.jelly [hunk 3]
```diff
@@ -24,7 +24,9 @@ THE SOFTWARE.
         <l:tasks>
             <l:task href="" icon="symbol-settings"
                     title="${app.hasPermission(app.MANAGE) ? '%Configure' : '%View Configuration'}"/>
-            <l:delete permission="${app.MANAGE}" title="${%Delete Pod Template}" message="${%delete.template(it.name)}"/>
+            <j:if test="${it.readonlyFromUi != true}">
```
Comment (Vlatombe) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/PodTemplate/sidepanel.jelly:27]:
```suggestion
            <j:if test="${!it.readonlyFromUi}">
```

**Review by Vlatombe — APPROVED** [review]
A few simplifications could be applied, but otherwise it looks good.

**Comment by Pldi23** [comment]
CI failure seems not related to the fix, might be a flake, @Vlatombe can you please retrigger the ci again as I don't have permissions?

**Pull request merged**